source/common/quic/envoy_quic_client_stream.cc

```diff
@@ -113,11 +113,10 @@ void EnvoyQuicClientStream::resetStream(Http::StreamResetReason reason) {
   Reset(envoyResetReasonToQuicRstError(reason));
 }
 
-void EnvoyQuicClientStream::switchStreamBlockState(bool should_block) {
-  if (should_block) {
+void EnvoyQuicClientStream::switchStreamBlockState() {
+  if (read_disable_counter_ > 0) {
     sequencer()->SetBlockedUntilFlush();
   } else {
-    ASSERT(read_disable_counter_ == 0, "readDisable called in between.");
     sequencer()->SetUnblocked();
```

Comment:
looking at SetBlockedUntilFlush it's pretty clear calling it extra times won't waste CPU or be unsafe.  If we read disable then read enable, we may end up calling some spurious SetUnblocked.  That _looks_ ok but can you confirm, and comment in code if it's fine?

Reply:
Yes, an extra OnDataAvailable() won't cause another push as data should have already been pushed in QUICHE stack or remain blocked by other condition.

Reply:
SG, can you please comment that in code then?  The other Envoy read enable/disable need parity (one unblock per block) so I think it's worth calling out that this is different.

Reply:
Commented at scheduling async_stream_blockage_change_ in readDisable() about the transient block and unblock.

Reply:
ah, that's not quite what I was looking for.
Can we do something like comment in switchStreamBlockState that unlike the envoy readDisable calls, that if the underlying QUIC code has received N block calls, it will still unblock on one block call (so parity between block and unblock calls is not necessary)?

Reply:
done